modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/SCIM2PaginationTestCase.java

```diff
@@ -0,0 +1,307 @@
+
+        JSONObject emailWork = new JSONObject();
+        emailWork.put(TYPE_PARAM, EMAIL_TYPE_WORK_ATTRIBUTE);
+        emailWork.put(VALUE_PARAM, workEmail);
+
+        JSONObject emailHome = new JSONObject();
+        emailHome.put(TYPE_PARAM, EMAIL_TYPE_HOME_ATTRIBUTE);
+        emailHome.put(VALUE_PARAM, homeEmail);
+
+        JSONArray emails = new JSONArray();
+        emails.add(emailWork);
+        emails.add(emailHome);
+
+        rootObject.put(EMAILS_ATTRIBUTE, emails);
+
+        rootObject.put(PASSWORD_ATTRIBUTE, password);
+
+        StringEntity entity = new StringEntity(rootObject.toString());
+        request.setEntity(entity);
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 201, "User has not been created successfully");
+
+        Object responseObj = JSONValue.parse(EntityUtils.toString(response.getEntity()));
+        EntityUtils.consume(response.getEntity());
+
+        String usernameFromResponse = ((JSONObject) responseObj).get(USER_NAME_ATTRIBUTE).toString();
+        assertEquals(usernameFromResponse, userName);
+
+        String userId;
+        userId = ((JSONObject) responseObj).get(ID_ATTRIBUTE).toString();
+        userIds.add(userId);
+        assertNotNull(userId);
+    }
+
+    @Test(dependsOnMethods = {"testUserListingPagination", "testUserListingPaginationWithoutParams"})
+    public void testDeleteUser() throws Exception {
+
+        for (String userId : userIds) {
+            deleteUser(userId);
+        }
+    }
+
+    private void deleteUser(String userId) throws IOException {
+
+        String userResourcePath = getUserPath() + "/" + userId;
+        HttpDelete request = new HttpDelete(userResourcePath);
+        request.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
+        request.addHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 204, "User has not been retrieved successfully");
+
+        EntityUtils.consume(response.getEntity());
+
+        userResourcePath = getUserPath() + "/" + userId;
+        HttpGet getRequest = new HttpGet(userResourcePath);
+        getRequest.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
+        getRequest.addHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+
+        response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 404, "User has not been deleted successfully");
+        EntityUtils.consume(response.getEntity());
+    }
+
+    private int getCurrentUsersCount() throws Exception {
+
+        return userMgtClient.getUserList().size();
+    }
+
+    private boolean isAllUsersExists(JSONArray response) throws Exception {
+
+        boolean usersExists = false;
+        HashSet<String> usersList = new HashSet<>();
+        for (Object user : response
+        ) {
```

Comment:
Wrong code formatting. Please use the wso2 coding style.

Reply:
added